core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -824,11 +825,15 @@ private[spark] class MapOutputTrackerWorker(conf: SparkConf) extends MapOutputTr
       endPartition: Int): Iterator[(BlockManagerId, Seq[(BlockId, Long, Int)])] = {
     logDebug(s"Fetching outputs for shuffle $shuffleId, mappers $startMapIndex-$endMapIndex" +
       s"partitions $startPartition-$endPartition")
-    val statuses = getStatuses(shuffleId, conf)
     try {
+      val statuses = getStatuses(shuffleId, conf)
```

Comment:
I'd prefer to add another try catch when we call `broadcast.vaule` at:

https://github.com/apache/spark/blob/a2aa966ef64bc06f65a646777568427d360605e9/core/src/main/scala/org/apache/spark/MapOutputTracker.scala#L969

and throw `FetchFailedException` when we catch `SparkException` from broadcast.